lib/Dialect/LWE/IR/NewLWEAttributes.td

```diff
@@ -348,4 +353,60 @@ def LWE_ModulusChainAttr : AttrDef<LWE_Dialect, "ModulusChain"> {
   // let genVerifyDecl = 1; // Verify index into list
 }
 
+def LWE_BVKeySwitchAttr : AttrDef<LWE_Dialect, "BVKeySwitch"> {
+  let mnemonic = "bv_keyswitch_technique";
+  let description = [{
+    An attribute describing the BV technique for keyswitch.
+
+    `base` is the radix base used in decomposition of the coefficient modulus
+    `Q` (non-RNS case) / `qi` (RNS case).
+
+    `dnum` is the number of large digits for the RNS case. It takes effect
+    only when `base` equals 0.
+
+    Check Appendix A of https://eprint.iacr.org/2021/204.pdf for more detail.
+  }];
+
+  let parameters = (ins
+    "IntegerAttr":$base,
+    "IntegerAttr":$dnum
```

Comment:
thank you for considering the RNS variants! I understand it's important to know the dnums so that the type converter knows the type / shape of the key switching key. For the lowering, we'll need the actual RNS moduli, which we'll pull from the RNS modulus of the ring in the keyswitching key, right?